crates/ruff_linter/src/linter.rs

```diff
@@ -197,9 +208,12 @@ pub fn check_path(
                 // to see if the diagnostic persists to the end of the function.
                 pycodestyle::rules::syntax_error(&mut diagnostics, &parse_error, locator);
                 error = Some(parse_error);
+                CommentRanges::default()
```

Comment:
I'm not sure if it makes sense to consume the `Program` and include `CommentRanges` to `LinterResult`. I'm thinking of just cloning the `ParseError` here, it shouldn't affect performance at all.

Reply:
Yeah we could do that too. Or you could consider adding a `take_errors` method to `Program`.